### PR TITLE
Implement a basic API

### PR DESCRIPTION
**Pull request by kchulj**
The basic structure of a `Faucet API` has been implemented.
Faucet should expose a basic API so it can be called from a frontend service (website) to make `Transaction`s.

`getUTXOs` performs a HTTP `GET /utxos` and returns an array of all UTXOs known.

Part of faucet #15

### CODE REVIEW

File: source/faucet/main.d [hunk 1]
```diff
@@ -84,145 +86,172 @@ private struct State
     }
 }
 
-/*******************************************************************************
+public class Faucet : IFaucet
```
Comment (AndrejMitrovic) [inline source/faucet/main.d:89]: Needs docs.

File: source/faucet/main.d [hunk 2]
```diff
@@ -84,145 +86,172 @@ private struct State
...
+{
+    /// Config instance
+    private Config config;
 
-    Splits the Outputs from `utxo_rng` towards `count` random keys
+    /// State instance
```
Comment (AndrejMitrovic) [inline source/faucet/main.d:94]: A brief description of what the state represents and when it may change would be nice.

Comment (AndrejMitrovic) [reply source/faucet/main.d:94]: Also a default initializer here would help make it clear what the initial state is.

File: source/faucet/main.d [hunk 3]
```diff
@@ -84,145 +86,172 @@ private struct State
...
+                                      this.config.count))
+            {
+                client.putTransaction(tx);
+                logDebug("Transaction sent: %s", tx);
+            }
+        }
+    }
+
+    /// GET: /utxos
+    public override UTXO[Hash] getUTXOs () pure nothrow @safe
```
Comment (AndrejMitrovic) [inline source/faucet/main.d:252]: Could you move this into a separate commit? Because this commit refactors the code and also adds a new feature.

**Review by AndrejMitrovic — CHANGES_REQUESTED** [review]

File: source/faucet/main.d (hunk 2 above)
Comment (kchulj) [reply source/faucet/main.d:94]: > A brief description of what the state represents and when it may change would be nice.

Should it be `///` or `/**` style ?

Comment (AndrejMitrovic) [reply source/faucet/main.d:94]: We usually use `///` for fields. You can have multiple lines of them, that's okay.

**Comment by kchulj** [comment]
Fixed. Thanks for the review.

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by AndrejMitrovic** [comment]
Thanks for your work!
